src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,269 @@
+// Alina TODO:
+//  - pagination and scroll
+// - submission
+
+import './DrugUploadTool.scss'
+
+import * as React from 'react'
+
+import { Link } from 'react-router-dom'
+
+import { Entity, } from '../../utils/jsonResponses/Entity'
+import { EntityLookupRequest } from '../../utils/jsonResponses/EntityLookupRequest'
+import { UserProfile } from '../../utils/jsonResponses/UserProfile'
+import { SynapseClient } from '../../utils'
+import { faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import WarningModal from './WarningModal'
+
+
+import _ from 'lodash';
+import moment from 'moment';
+
+type UserData = {
+  fileList: Entity[];
+  existingData: any;
+  dataFolderId: string;
+}
+
+export type UserFileGridProps = {
+  token?: string;
+  parentContainerId: string; //projectId
+  pathpart: string;
+}
+
+type UserFileGridState = {
+  token?: string;
+  userprofile?: any;
+  dataFolderId?: string,
+  fileList: Entity[]
+  isLoading: boolean;
+  modalContext?: { 'action': Function, 'arguments': any[] }
+}
+
+
+export default class UserFileGrid extends React.Component<UserFileGridProps, UserFileGridState> {
+  modalTitle='Trash Submission?' 
+  modalCopy=<><p>This submission is currently incomplete and has not been submitted. If you trash this submission, you won't be able to recover the data.</p>
+  <p>Are you sure you want to trash this submission?' </p></>
+  
+  
+  constructor(props: UserFileGridProps) {
+    super(props)
+    this.state = {
+      isLoading: true,
+      token: this.props.token,
+      fileList: []
+    }
+  }
+
+  componentDidMount(): Promise<any> {
+    // Note:  All portals should do this once on the initial app load.
+    // This looks for the session token cookie (HttpOnly, unable to directly access), and initialize the session if it does exists.
+   //SynapseClient.detectSSOCode()
+    return SynapseClient.getSessionTokenFromCookie()
+      .then((sessionToken: any) => { return this.refresh(sessionToken) })
+      .catch((error: any) => {
+        console.error(error)
+      })
+
+      this.refresh(this.state.token);
+  }
+
+
+  refresh = (token?: string): Promise<any> => {
+    this.setState({ token })
+    if (token) {
+      return this.getUserFileListing(token!, this.props.parentContainerId).catch((error) => {
+        this.onError(error)
+      })
+
+    } else {
+      return Promise.reject()
+    }
+  }
+
+
+
+  private getUserFileListing = async (token: string, parentContainerId: string): Promise<UserData> => {
+    this.setState({ isLoading: true });
+    const result: UserData = {
+      fileList: [],
+      existingData: {},
+      dataFolderId: ''
+    }
+
+    // get user profile
+    const userprofile = await SynapseClient.getUserProfile(token)
```

Comment:
camelCase on userProfile, and this could get wrapped in a try/catch -- Although it really shouldn't fail, I think it doesn't hurt to have more error catching.